File: src/Web/WebAngular/src/app/domain/commands/connector-function-history-commands/create-connector-function-history.command.ts
import { CreateConnectorFunctionInputCommand } from "./create-connector-function-input.command";

export interface CreateConnectorFunctionHistoryCommand {
	connectorFunctionId: string;
	script: Buffer | Uint8Array | string;
	package: Buffer | Uint8Array | string;
	inputs: CreateConnectorFunctionInputCommand[];
}

File: src/Core/Houston.Application/CommandHandlers/ConnectorFunctionHistoryCommandHandlers/Create/CreateConnectorFunctionHistoryCommand.cs
namespace Houston.Application.CommandHandlers.ConnectorFunctionHistoryCommandHandlers.Create {
	public sealed record CreateConnectorFunctionHistoryCommand(Guid ConnectorFunctionId, byte[] Script, byte[] Package, string Version, List<CreateConnectorFunctionInputCommand> Inputs) : IRequest<IResultCommand>;

	public sealed record CreateConnectorFunctionInputCommand(InputType InputType, string Name, string Placeholder, string Replace, bool Required, string? DefaultValue, string[]? Values, bool AdvancedOption);
}
